src/pipecat/services/ultravox.py

```diff
@@ -362,18 +368,20 @@ async def _process_audio_buffer(self) -> AsyncGenerator[Frame, None]:
                     await self.stop_processing_metrics()
                     
                     logger.info(f"Generated text: {full_response}")
-                    
                     # Create a transcription frame with the generated text
-                    transcription = full_response.strip()
-                    if transcription:
-                        yield TranscriptionFrame(
-                            text=transcription,
-                            interim_text="",
-                            timestamp=time_now_iso8601()
-                        )
-                    else:
-                        logger.warning("Empty transcription result")
-                        yield ErrorFrame("Empty transcription result")
+                    sentences = []
+                    current = ""
+                    for char in full_response.strip():
+                        current += char
+                        if char in '.!?':
+                            sentences.append(current.strip())
+                            current = ""
+                    if current.strip():  # Add any remaining text
+                        sentences.append(current.strip())
+
+                    for sentence in sentences:
+                        text_frame = TextFrame(text=sentence)
```

Comment:
I believe this should be a `TranscriptionFrame`

Reply:
Oh, sorry, this needs to be `LLMTextFrame`. Also, you need to add this:

```
if sentences:
    yield LLMFullResponseStartFrame()

for sentence in sentences:
     text_frame = LLMTextFrame(text=sentence)
    yield text_frame

if sentences:
    yield LLMFullResponseEndFrame()
```

Reply:
Thinking more about it, I believe we probably don't need all the new code and we just need the following:

```
full_response = full_response.strip()
if full_response:
    yield LLMFullResponseStartFrame()
    yield LLMTextFrame(text=full_response)
    yield LLMFullResponseEndFrame()
```